CAIPs/caip-25.md

```diff
@@ -39,64 +39,136 @@ forthcoming one.
 Within that session model, this interface outlines the authorization of an
 injected provider per namespace.
 
+The application interfaces with a provider to populate a session with a base
+state describing authorized chains, methods, event, and accounts.  This
+negotation takes place by sending the application's REQUIRED and REQUESTED
+properties of the session. If any requirements are not met, a failure response
+expressive of one or more specific failure states will be sent (see below).
+Conversely, a succesful response will contain all the required properties *and
+the provider's choice of the optional properties* expressed as a unified set of
+parameters.
+
 ### Request
 
-The application would interface with a provider to authorize a provider with a
-given set of parameters by calling the following JSON-RPC request
+Example:
 
-```
+```jsonc
 {
-    "id": 1,
-    "jsonrpc": "2.0",
-    "method": "provider_authorization",
-    "params": {
-        "eip155": {
-            "chains": ["eip155:1"],
-            "methods": ["eth_sendTransaction", "eth_signTransaction", "eth_sign", "personal_sign"]
-            "events": ["accountsChanged", "chainChanged"]
-        },
-        "cosmos": {
-            ...
-        }
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "provider_authorization",
+  "params": {
+    "required": {
+      "eip155": {
+        "chains": ["eip155:1", "eip155:137"],
+        "methods": ["eth_sendTransaction", "eth_signTransaction", "eth_sign", "get_balance", "personal_sign"],
+        "events": ["accountsChanged", "chainChanged"]
+      },
+      "eip155:10": {
+        "methods": ["get_balance"],
+        "events": ["accountsChanged", "chainChanged"]
+      },
+      "cosmos": {
+        ...
+      }
     },
-
+    "optional":{
+      "eip155:42161": {
+        "methods": ["eth_sendTransaction", "eth_signTransaction", "get_balance", "personal_sign"],
+        "events": ["accountsChanged", "chainChanged"]
+      },
+      "sessionProperties": {
+        "expiry": "2022-12-24T17:07:31+00:00",
```

Comment:
I would put this in the response instead of the request

IMO it should always be the responder that defines it and/or calculates it

Alternatively there should be an upper bond for the expiry (eg. 30 days)

Reply:
Hehe, scroll down, in my example the responder picked a 7-day expiry and overrode the request :D